server/search/hits.go

```diff
@@ -112,3 +104,83 @@ func NewSearchHit(tsHit *tsApi.SearchResultHit) *Hit {
 		},
 	}
 }
+
+func fromHighlights(highlights []tsApi.SearchHighlight, fields *[]*api.MatchField) {
+	for _, f := range highlights {
+		name := ""
+		if f.Field != nil {
+			name = *f.Field
+		}
+		*fields = append(*fields, &api.MatchField{
+			Name: name,
+		})
+	}
+}
+
+func fromHighlight(highlight map[string]interface{}, fields *[]*api.MatchField) {
+	// check first in highlight
+	for name, value := range highlight {
+		if mp, ok := value.(map[string]any); ok {
+			// any non array match should just fall in this "if"
+			if isMatchedToken(mp) {
+				*fields = append(*fields, &api.MatchField{
+					Name: name,
+				})
+			}
+			continue
+		}
+
+		if mArr, ok := value.([]any); ok {
+			for _, each := range mArr {
+				if mp, ok := each.(map[string]any); ok {
+					matchedToken := isMatchedToken(mp)
+					if matchedToken {
+						// simple array
+						*fields = append(*fields, &api.MatchField{
+							Name: name,
+						})
+						break
+					}
+
+					// now this means we can have an array of objects, we iterate one level
+					// to build matched fields
+					buildMatchedForArrObjects(name, mp, fields)
+				}
+			}
+		}
+	}
+}
```

Comment:
newline missing?

Reply:
Saw it now, will fix it in next diff.

Reply:
I wonder how it passed the lints.
We need to harden it more :)

Reply:
I don't think lint complains about the extra line? Yes, we need to add that check as well.